staging/src/k8s.io/apimachinery/pkg/util/net/http_test.go

```diff
@@ -895,6 +895,11 @@ func TestIsProbableEOF(t *testing.T) {
 			err:      fmt.Errorf("error foo - Use of closed network connection - error bar"),
 			expected: true,
 		},
+		{
+			name:     "with http2 client connection lost error",
+			err:      fmt.Errorf("http2: client connection lost"),
+			expected: true,
```

Comment:
This test doesn't address Jordan's concern. We need to actually trigger a roundtrip that is closed due to unhealthy connection, and verify the error message is indeed "http2: client connection lost". I think you can modify this [unit test](https://github.com/kubernetes/kubernetes/blob/19c8a2127177b43effb9bfe1de272d6f08ea56e8/staging/src/k8s.io/client-go/rest/connection_test.go#L96) to do the described test.